### ISSUE TITLE
Question: What do i need to do to move an entity into a different c# namespace?

### ISSUE THREAD

**Issue by automatonic** [dotnet/efcore#6615]
Say I have an entity that looks like this:

```
namespace OldNamespace
{
    [Table("Widget", Schema = "summat")]
    public class Widget
    {
    }
}
```

And I want to move it to a `NewNamespace` in a C# sense (but not in a database table sense):

```
namespace NewNamespace
{
    [Table("Widget", Schema = "summat")]
    public class Widget
    {
    }
}
```

The model snapshot that is generated includes the full c# namespace of the entity, so it seems problematic that EF will know what an old/existing namespace will be after a migration?

```
        protected override void BuildModel(ModelBuilder modelBuilder)
        {
              //...
                modelBuilder.Entity("OldNamespace.Widget", b =>
                {
```
### Further technical details

EF Core version: 1.0.1
Operating system: Win10
Visual Studio version: VS 2015.3

**Comment by automatonic** [comment]
Note that this would be a move of ALL entities and the db context. What do I do for old migrations? The snapshot?

**Comment by automatonic** [comment]
If preferable, i have also [asked on stackoverflow](http://stackoverflow.com/questions/39729009/how-do-i-move-an-existing-entity-into-a-different-c-sharp-namespace-and-maintain).

**Comment by automatonic** [comment]
Once i finally go the entity namespaces refactored, this went off without a hitch. I refactored the entities, dbContext, and migrations to all be in the new namespace. I was able to add a new migration that ran against existing db state, and I was able to update a new database from scratch without issue. It just works, and the reason i was being cautious was because of an issue i ran into along these lines in the EF7 RC1 code base (which may have actually stemmed from a find/replace of my connection strings, in which the database name matched the refactored Namespace).
